python-avd/pyavd/_eos_designs/structured_config/underlay/route_maps.py

```diff
@@ -141,6 +142,46 @@ def route_maps(self: AvdStructuredConfigUnderlay) -> list | None:
                 ]
                 route_maps.append({"name": "RM-BGP-UNDERLAY-PEERS-OUT", "sequence_numbers": sequence_numbers})
 
+        for neighbor in self.shared_utils.l3_interfaces_bgp_neighbors:
+            if prefix_list_in := get(neighbor, "ipv4_prefix_list_in"):
+                # RM-BGP-<PEER-IP>-IN
+                sequence_numbers = [
+                    {
+                        "sequence": 10,
+                        "type": "permit",
+                        "match": [f"ip address prefix-list {prefix_list_in}"],
+                    },
+                ]
+                # set no advertise is set only for wan neighbors, which will also have
+                # prefix_list_in
+                if neighbor.get("set_no_advertise"):
+                    sequence_numbers[0]["set"] = ["community no-advertise additive"]
+
+                route_maps.append({"name": neighbor["route_map_in"], "sequence_numbers": sequence_numbers})
+
+            if prefix_list_out := get(neighbor, "ipv4_prefix_list_out"):
+                sequence_numbers = [
+                    {
+                        "sequence": 10,
+                        "type": "permit",
+                        "match": [f"ip address prefix-list {prefix_list_out}"],
+                    },
+                    {
+                        "sequence": 20,
+                        "type": "deny",
+                    },
+                ]
+            else:
+                # RM-BGP-<PEER-IP>-OUT
```

Comment:
Just moving the comment.
```suggestion
            # RM-BGP-<PEER-IP>-OUT
            if prefix_list_out := get(neighbor, "ipv4_prefix_list_out"):
                sequence_numbers = [
                    {
                        "sequence": 10,
                        "type": "permit",
                        "match": [f"ip address prefix-list {prefix_list_out}"],
                    },
                    {
                        "sequence": 20,
                        "type": "deny",
                    },
                ]
            else:
```